### PR TITLE
docs(Table): add Editable Table example

### LINKED ISSUE #6597: Add example: Composable Table with editable rows

### PR DESCRIPTION
**Pull request by adamviktora**
**What**: Closes #6597

### CODE REVIEW

**Review by wise-king-sullyman — COMMENTED** [review]
Looks like the example is erroring out for some reason on the surge build, but it works locally, not sure what's going on there.

What's the deal with importing each react-core component from dist individually? I tried changing that to a normal import locally and it didn't seem to cause any issues.

I do also wonder if this should be a demo instead of an example with how many react-core components it's using 🤔 

Other than those questions though this looks awesome 🚀

**Comment by adamviktora** [comment]
The example error in the surge build is probably because the surge build is old (I then force pushed another commit), and I was using a generic component in the previous commit and it could not parse the syntax.

About the individual imports from dist, I at first thought we should import each file individually when importing to another package, the same way we import icons (I read [this](https://medium.com/@marvusm.mmi/webpack-module-federation-think-twice-before-sharing-a-dependency-18b3b0e352cb) article from Martin, but I misunderstood it).

**Review by wise-king-sullyman — APPROVED** [review]
🔥

**Review by nicolethoen — CHANGES_REQUESTED** [review]
when using a keyboard to edit the rows, the browser focus gets lost when toggling between edit and non-edit states since the buttons disappear. We might need to move focus intentionally when you use keyboard to click pencil icon or save/cancel icons.

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Couple of comments below. Additionally, there's some markup issues with the table itself:

- There needs to be a `Th` component for the editable buttons column. If we want to keep it blank, then screenreadertext needs to be supplied such as "Row edit actions". See the [table with actiosn example](https://www.patternfly.org/components/table#actions)
- The edit buttons (edit, save, close) need to be wrapped in a `Td`. Right now the `div`  wrappers are rendered as chlldren to `tr`

Both of the above issues cause the edit buttons not to be navigable via screen reader since semantically there's no valid table cells to navigate to.

File: packages/react-table/src/components/Table/examples/TableEditable.tsx [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            aria-label={`${columnNames.textInputDisabled} ${ariaLabel}`}
+            isDisabled={true}
+            value={editedData.textInputDisabled ?? ''}
+          />
+        }
+      />
+      <EditableCell
+        dataLabel={columnNames.checkboxes}
+        staticValue={data.checkboxes.join(', ')}
+        editingValue={dataOptions?.checkboxes.map((option) => {
```
Comment (thatblindgeye) [inline packages/react-table/src/components/Table/examples/TableEditable.tsx:158]: Only other thing is that the checkboxes and radios for each row should be wrapped in a `pf-v5-c-inline-edit__group pf-m-column` element, with the checkbox group having `role="group"` and radios having `role="radiogroup"`. Each should also have an aria-label to the group container. Core markup (core needs to be updated to add the role and aria label to the checkbox group):

image

**Review by thatblindgeye — APPROVED** [review]
Other than the lint error lgtm 😎

**Pull request merged**